### PR TITLE
FORD-68/make it possible to purchase videos

### PR DESCRIPTION
**Pull request by Xayer**

### CODE REVIEW

File: js/bp-cl-oauth-login.js [hunk 1]
```diff
@@ -87,9 +87,25 @@ function setDownloadUrl(downloadBtns, response)
...
-                btnAfterResponse.setAttribute('href', btn['data-response']);
-                btnAfterResponse.setAttribute('target', '_blank');
-                btnAfterResponse.removeAttribute('data-toggle');
+                var btnType = btnAfterResponse.getAttribute('data-type');
+                //allDisclaimerBtns.setAttribute('href', btn['data-response']);
+                if( btnType == 'video'){
+                    videoModalTarget = btnAfterResponse.getAttribute('data-target');
+                    btnAfterResponse.setAttribute('data-target',  videoModalTarget +'-video');
+                    var videoModals = document.getElementById(videoModalTarget.replace(/^#/, '')+'-video');
+                    console.dir(videoModals);
```
Comment (MFlor) [inline js/bp-cl-oauth-login.js]: Console.dir!

File: src/Bonnier/WP/ClOauth/Http/Routes/OauthLoginRoute.php [hunk 2]
```diff
@@ -192,8 +192,12 @@ public function has_access(WP_REST_Request $request)
                         break;
                     case 'video' :
                         $result[$key]['data-id'] = $widgetBtn['data-id'];
-                        //TODO finish Video implementation
-                        $result[$key]['data-response'] = $widgets[$widgetBtn['data-index']]['embed_url'];
+                        preg_match("/src=\"(.*?)\"/", $widgets[$widgetBtn['data-index']]['embed_url'], $matches);
+                        /*if(!empty($matches)){
```
Comment (MFlor) [inline src/Bonnier/WP/ClOauth/Http/Routes/OauthLoginRoute.php]: Maybe remove comment block?

File: src/Bonnier/WP/ClOauth/Http/Routes/OauthLoginRoute.php [hunk 3]
```diff
@@ -173,7 +173,7 @@ public function has_access(WP_REST_Request $request)
         $data = json_decode($request->get_param('data'), true);
 
         if(!$this->clRepo->hasAccessTo($purchaseId) || !$data) {
-            return new WP_REST_Response(['status' => 'No Access']);
+            //return new WP_REST_Response(['status' => 'No Access']);
```
Comment (MFlor) [inline src/Bonnier/WP/ClOauth/Http/Routes/OauthLoginRoute.php]: Comments?

File: js/bp-cl-oauth-login.js [hunk 4]
```diff
@@ -87,9 +87,25 @@ function setDownloadUrl(downloadBtns, response)
             }
             else
             {
-                btnAfterResponse.setAttribute('href', btn['data-response']);
-                btnAfterResponse.setAttribute('target', '_blank');
-                btnAfterResponse.removeAttribute('data-toggle');
+                var btnType = btnAfterResponse.getAttribute('data-type');
+                //allDisclaimerBtns.setAttribute('href', btn['data-response']);
```
Comment (MFlor) [inline js/bp-cl-oauth-login.js]: Also comment?

**Comment by Xayer** [comment]
@MFlor ¯\_(ツ)_/¯

**Pull request merged**